wrapplog.py

```diff
@@ -1,25 +1,71 @@
-import collections
-import inspect
+import os
 import json
-import structlog
+import inspect
+import collections
+from datetime import datetime
 
+import structlog
+from structlog.processors import JSONRenderer
+from structlog import wrap_logger, BoundLogger, PrintLogger, PrintLoggerFactory
 
 
+# Deprecated
 def start_logging(output=None):
-    structlog.configure(
-            logger_factory=structlog.PrintLoggerFactory(output),
-            processors=[render_wrapp_log])
+    return
+
+
+class NumberValueError(Exception):
+    pass
+
+
+class WrappPrintLogger(PrintLogger):
+    def msg(self, message):
+        return super(WrappPrintLogger, self).msg(message)
+
+    event = metric = msg
+
+
+class WrappLogger(BoundLogger):
+    def event(self, name, data=None):
+        return super(WrappLogger, self)._proxy_to_logger('event', name, data=data)
+
+    def metric(self, name, value=1):
+        try:
+            float(value)
+        except ValueError:
+            raise NumberValueError(value)
+        return super(WrappLogger, self)._proxy_to_logger('metric', name, value=value)
 
 
 class Logger(object):
-    def __init__(self, namespace=None, source=None):
+    def __init__(self, output=None,
+                 namespace=None, source=None,
+                 service=None, host=None):
+        log = wrap_logger(
+                        WrappPrintLogger(output),
+                        wrapper_class=WrappLogger,
+                        processors=[
+                            add_timestamp,
+                            order_fields,
+                            JSONRenderer(),
+                            render_wrapp_log,
+                        ])
+
         if not namespace:
             try:
                 frame = inspect.currentframe(1)
                 namespace = frame.f_globals['__name__']
             except:
                 namespace = 'unknown'
-        self._log = structlog.get_logger(namespace=namespace)
+        self._log = log.bind(namespace=namespace)
```

Comment:
You're right. It won't. I will fix this and exclude it for event and metric logs.